apps/gitness/src/pages/repo-sandbox/repo-sandbox-files.tsx

```diff
@@ -44,73 +44,79 @@ interface SidebarProps {
-    setIsOpen,
-    selectedBranch,
-    selectBranch,
-    branchList,
-    navigateToNewFile,
-    navigateToFile,
-    query,
-    handleInputChange,
-    fileText,
-    filesList,
-    repoDetails
-  }) => {
-    return (
-      <div className="flex flex-col gap-5">
-        <div className="w-full grid grid-cols-[1fr] auto-cols-auto grid-flow-col gap-3 items-center">
-          {branchList && (
-            <BranchSelector size="sm" name={selectedBranch} branchList={branchList} selectBranch={selectBranch} />
-          )}
-          <ButtonGroup.Root
-            spacing="0"
-            className="shadow-border shadow-[inset_0_0_0_1px] rounded-md h-full overflow-hidden">
-            <Button size="sm" variant="ghost" className="border-l rounded-none p-0 w-8" onClick={navigateToNewFile}>
-              <Icon size={15} name="add-file" className="text-primary/80" />
-            </Button>
-          </ButtonGroup.Root>
-        </div>
-        <Popover open={isOpen} onOpenChange={setIsOpen}>
-          <PopoverTrigger asChild>
-            <Button variant="outline" role="combobox" aria-expanded={isOpen} className="justify-between">
-              {'Search...'}
-              <Icon name="chevron-down" size={12} className="chevron-down " />
-            </Button>
-          </PopoverTrigger>
-          <PopoverContent className="w-[600px] p-0" align="start">
-            <Command>
-              <CommandInput placeholder="Search files..." className="h-9" onInput={handleInputChange} value={query} />
-              <CommandList>
-                <CommandEmpty>No file found.</CommandEmpty>
-                <CommandGroup>
-                  {filesList?.map((file: string, idx: number) => (
-                    <CommandItem
-                      key={idx}
-                      value={file}
-                      onSelect={() => {
-                        if (file) {
-                          navigateToFile(file)
-                          setIsOpen(false)
-                        }
-                      }}>
-                      {fileText(file)}
-                    </CommandItem>
-                  ))}
-                </CommandGroup>
-              </CommandList>
-            </Command>
-          </PopoverContent>
-        </Popover>
-        {repoDetails?.content?.entries?.length && (
-          <Explorer repoDetails={repoDetails} selectedBranch={selectedBranch} />
+const Sidebar: React.FC<SidebarProps> = ({
+  isOpen,
+  setIsOpen,
+  selectedBranch,
+  selectBranch,
+  branchList,
+  navigateToNewFile,
+  navigateToFile,
+  query,
+  handleInputChange,
+  fileText,
+  filesList,
+  repoDetails
+}) => {
+  return (
+    <div className="flex flex-col gap-5">
+      <div className="w-full grid grid-cols-[1fr] auto-cols-auto grid-flow-col gap-3 items-center">
+        {branchList && (
+          <BranchSelector size="sm" name={selectedBranch} branchList={branchList} selectBranch={selectBranch} />
         )}
+        <ButtonGroup.Root
```

Comment:
majority of this code should be in the playground.. not a blocker, but let's think about moving the presentation part of the UI there